c++/src/Writer.cc

```diff
@@ -307,6 +307,8 @@ namespace orc {
     static const char* magicId;
     static const WriterId writerId;
     bool useTightNumericVector;
+    long stripesAtLastFlush;
```

Comment:
If I use `size_t` my compiler complains:
```
error: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Werror=sign-compare]
  412 |     if (stripesAtLastFlush != fileFooter.stripes_size()) {
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

`Footer::stripes_size()` returns `int`, instead of casting to `size_t` maybe we could just do: `int stripesAtLastFlush;`?

Reply:
We can apply static_cast here. Something like `if (stripesAtLastFlush != static_cast<size_t>(fileFooter.stripes_size())) `. Or we can make it int32_t here because `fileFooter.stripes_size()` is of type `int`.